framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1345,7 +1345,11 @@ private AnnotatedTypeMirror fromExpression(ExpressionTree tree) {
 
         AnnotatedTypeMirror result = TypeFromTree.fromExpression(this, tree);
 
-        if (shouldCache) {
+        if (shouldCache && tree.getKind() != Tree.Kind.NEW_CLASS) {
+            // Don't cache types of NEW operator
+            // Cached NEW_CLASS types may contain "Unknowned*" annotations
+            // Such annotations are from previous dataflow analysis
```

Comment:
@abslamp Can you update this PR and change this comment as suggested in https://github.com/typetools/checker-framework/pull/2643#discussion_r343972843